### ISSUE TITLE
Feature Request: Support validation of requests and responses based on openapi schema

### ISSUE THREAD

**Issue by kazkansouh** [nestjs/swagger#1468]
## I'm submitting a...
<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Feature request

It would be nice to have an option to validate requests and responses against the generated openapi schema.

I don't think this would be much work as other projects already provide this functionality and it should just be a matter of including them. For example, `express-openapi-validator` already provides [an integration with `nestjs`][validator]. I have lightly tested the code and its appears to work well with schemas generated by this project.

Further, this approach provides an alternative to class-validator (supported by nestjs) which is less mandraulic for basic classes/types. E.g. I dont need to decorate a number field with `@IsNumber`, the swagger cli plugin picks up those details automatically.

Also, it supports more advanced concepts such as `oneOf`.

### Example Middleware

Below is an example how it can be included as a middleware (largely based on the above repo). The only issue is that I set the document with a `static` method. This needs to be done before `app.listen(...)` is called. But I am sure people that understand the framework better than me can find a better solution - probably by injecting it.

```typescript
import { Injectable, NestMiddleware } from '@nestjs/common';
import * as OpenApiValidator from 'express-openapi-validator';
import {
  OpenApiRequestHandler,
  OpenAPIV3,
} from 'express-openapi-validator/dist/framework/types';

@Injectable()
export class OpenAPIValidatorMiddleware implements NestMiddleware {
  private static middlewares: OpenApiRequestHandler[] = [];

  use(req: any, res: any, rootNext: (err?: any) => void) {
    let idx = 0;
    next();

    function next(err?: any) {
      if (err || idx >= OpenAPIValidatorMiddleware.middlewares.length) {
        return rootNext(err);
      }
      return OpenAPIValidatorMiddleware.middlewares[idx++](req, res, next);
    }
  }

  static setup(document: OpenAPIV3.Document) {
    this.middlewares = OpenApiValidator.middleware({
      apiSpec: document,
      validateRequests: {
        allowUnknownQueryParameters: true,
        coerceTypes: false,
      },
      validateResponses: true,
      validateFormats: 'full',
    });
  }
}
```

and in the `main.ts` file:

```typescript
const document = SwaggerModule.createDocument(app, config);
SwaggerModule.setup('swagger', app, document);
OpenAPIValidatorMiddleware.setup(document as OpenAPIV3.Document);
await app.listen(3000);
```

[validator]: https://github.com/cdimascio/express-openapi-validator/tree/master/examples/9-nestjs

**Comment by kamilmysliwiec** [comment]
Please search through some of our old issues on this (this has been discussed several times in the past).
